api/src/main/java/graphql/nadel/schema/OverallSchemaGenerator.java

```diff
@@ -37,35 +37,37 @@
 @Internal
 public class OverallSchemaGenerator {
 
-    public GraphQLSchema buildOverallSchema(List<DefinitionRegistry> serviceRegistries, DefinitionRegistry commonTypes, WiringFactory wiringFactory) {
+    public GraphQLSchema buildOverallSchema(List<NadelDefinitionRegistry> serviceRegistries, WiringFactory wiringFactory) {
         SchemaGenerator schemaGenerator = new SchemaGenerator();
         RuntimeWiring runtimeWiring = RuntimeWiring.newRuntimeWiring()
-                .wiringFactory(wiringFactory)
-                .build();
-        return schemaGenerator.makeExecutableSchema(createTypeRegistry(serviceRegistries, commonTypes), runtimeWiring);
+            .wiringFactory(wiringFactory)
+            .build();
+        return schemaGenerator.makeExecutableSchema(createTypeRegistry(serviceRegistries), runtimeWiring);
     }
 
-    private TypeDefinitionRegistry createTypeRegistry(List<DefinitionRegistry> serviceRegistries, DefinitionRegistry commonTypes) {
-        //TODO: this merging not completely correct for example schema definition nodes are not handled correctly
-        Map<OperationKind, List<FieldDefinition>> fieldsMapByType = new LinkedHashMap<>();
-        Arrays.stream(OperationKind.values()).forEach(
-                value -> fieldsMapByType.put(value, new ArrayList<>()));
+    private TypeDefinitionRegistry createTypeRegistry(List<NadelDefinitionRegistry> serviceRegistries) {
+        // TODO: this merging not completely correct for example schema definition nodes are not handled correctly
```

Comment:
Is this comment still valid? @bbakerman you wrote it in

Reply:
kinda - but maybe its not worth it

for example

```
schema {
   query : Q
}
```
is technically valid SDL - but should we merge it / accept it in Nadel   say?  Its kinda pointless

I think also at the time I wrote this we didnt support `extend type Query` but thats changed I think

Reply:
Ah right, yea I think this is pretty silly. We do support schemas with

```
schema {
  query: Q
}
```

though. We just pick up to treat `Q` like `Query` and but the type "rename" is ignored.